x-pack/platform/packages/shared/kbn-classic-stream-flyout/src/components/create_classic_stream_flyout/steps/name_and_confirm/confirm_template_details_section.tsx

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useEffect, useState } from 'react';
+import {
+  EuiPanel,
+  EuiSpacer,
+  EuiTitle,
+  EuiDescriptionList,
+  useEuiTheme,
+  EuiBadge,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHealth,
+  EuiLoadingSpinner,
+  euiFontSize,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { css } from '@emotion/react';
+import type { TemplateDeserialized } from '@kbn/index-management-plugin/common/types';
+import {
+  formatDataRetention,
+  getPhaseDescriptions,
+  type IlmPolicyFetcher,
+  type PhaseDescription,
+} from '../../../../utils';
+
+interface ConfirmTemplateDetailsSectionProps {
+  template: TemplateDeserialized;
+  getIlmPolicy?: IlmPolicyFetcher;
+}
+
+export const ConfirmTemplateDetailsSection = ({
+  template,
+  getIlmPolicy,
+}: ConfirmTemplateDetailsSectionProps) => {
+  const euiThemeContext = useEuiTheme();
+  const { euiTheme } = euiThemeContext;
+
+  const [ilmPhases, setIlmPhases] = useState<PhaseDescription[] | null>(null);
+  const [isLoadingIlmPolicy, setIsLoadingIlmPolicy] = useState(false);
+
+  const ilmPolicyName = template.ilmPolicy?.name;
+
+  const phaseColors = useMemo(
+    () => ({
+      hot: euiTheme.colors.vis.euiColorVis6,
+      warm: euiTheme.colors.vis.euiColorVis9,
+      cold: euiTheme.colors.vis.euiColorVis2,
+      frozen: euiTheme.colors.vis.euiColorVis4,
+    }),
+    [euiTheme]
+  );
+
+  // Fetch ILM policy details when policy name is available
+  useEffect(() => {
+    if (!ilmPolicyName || !getIlmPolicy) {
+      setIlmPhases(null);
+      return;
+    }
+
+    const abortController = new AbortController();
+    setIsLoadingIlmPolicy(true);
+
+    getIlmPolicy(ilmPolicyName, abortController.signal)
+      .then((policy) => {
+        if (!abortController.signal.aborted && policy) {
+          const phases = getPhaseDescriptions(policy.policy.phases, phaseColors);
+          setIlmPhases(phases);
+        }
+      })
+      .catch((error) => {
+        // Silently fail - we'll just not show phases
```

Comment:
what if user assumes that policy has no phases, instead of that it's a failure. Maybe we should show a warning toast or error toast https://eui.elastic.co/docs/components/display/toast/#warning ?